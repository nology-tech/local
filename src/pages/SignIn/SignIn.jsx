import Layout from "../../components/Layout/Layout";
import logo from "../../assets/images/AppLogo.svg";
import TextField from "../../components/TextField/TextField";
import Button from "../../components/Button/Button";
import "./SignIn.scss";

const SignIn = () => {
  return (
    <Layout>
      <div className="sign-in">
        <div className="sign-in__intro">
          <img className="sign-in__logo" src={logo} alt="App Logo" />
          <h1>LOCAL</h1>
          <p>Bringing local business to local people</p>
        </div>
        <div className="sign-in__input-field">
          <TextField
            uniqueInputFieldName="signinUserName"
            inputType="text"
            labelHeader="Username"
            modifier="signin"
          />
          <TextField
            uniqueInputFieldName="signinPassword"
            inputType="text"
            labelHeader="Password"
            modifier="signin"
          />
          <div className="sign-in__buttons">
            <Button buttonText={"Sign in"} buttonName="signin" />
            <Button buttonText={"Sign in with Google"} buttonName="signin" />
          </div>
        </div>
      </div>
    </Layout>
  );
};

export default SignIn;
