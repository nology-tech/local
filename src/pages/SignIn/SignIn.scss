@use "../../styles/variables" as *;

.sign-in {
    display: flex;
    flex-direction: column;
    align-items: center;
    height: 100%;

    &__intro {
        display: flex;
        flex-direction: column;
        justify-content: flex-end;
        align-items: center;
        text-align: center;
        font-size: 1.3rem;
        margin-top: 10%;
        // height: 30%;
        // width: 80%;
    }

    &__logo {
        width: 50%;
        height: 55%;
    }

    &__input-field {
        border-style: solid;
        border-radius: 2rem;
        margin: 1rem auto;
        width: 90vw;
        height: 90vh;
    }

    &__buttons {
        // width: 100%;
        margin: 1rem auto;
        width: 90vw;
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-content: center;
    }

}