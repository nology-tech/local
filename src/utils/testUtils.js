import { jest } from "@jest/globals";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import UserContext from "../context/UserContext";
import mockUser from "../data/mockUser";

export const customRender = (
  ui,
  useRouting = true,
  contextOptions = {
    useContext: true,
    value: { user: mockUser, setUser: jest.fn() },
  }
) => {
  // wrap components in routing and context if requested
  const routingUiResult = useRouting && wrapWithRouting(ui);
  const contextUiResult =
    contextOptions.useContext &&
    wrapWithContext(routingUiResult, contextOptions.value);

  // use RTL's render function to return the test component
  return render(contextUiResult);
};

const wrapWithRouting = (ui) => {
  return <Router>{ui}</Router>;
};

const wrapWithContext = (ui, value) => {
  return <UserContext.Provider value={value}>{ui}</UserContext.Provider>;
};
