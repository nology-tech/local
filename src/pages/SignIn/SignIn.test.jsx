import { customRender } from "../../utils/testUtils";
import SignIn from "./SignIn";

it("should render the text on the display", () => {
  const { container } = customRender(<SignIn />);

  expect(container).toMatchSnapshot();
});
