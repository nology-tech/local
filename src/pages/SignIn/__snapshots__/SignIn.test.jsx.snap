// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the text on the display 1`] = `
<div>
  <div
    class="layout"
  >
    <div
      class="sign-in"
    >
      <div
        class="sign-in__intro"
      >
        <img
          alt="App Logo"
          class="sign-in__logo"
          src="AppLogo.svg"
        />
        <h1>
          LOCAL
        </h1>
        <p>
          Bringing local business to local people
        </p>
      </div>
      <div
        class="sign-in__input-field"
      >
        <div
          class="text-field text-field--signin"
        >
          <label
            class="text-field__label text-field__label--signin"
          >
            Username
          </label>
          <input
            class="text-field__input text-field__input--signin"
            type="text"
          />
        </div>
        <div
          class="text-field text-field--signin"
        >
          <label
            class="text-field__label text-field__label--signin"
          >
            Password
          </label>
          <input
            class="text-field__input text-field__input--signin"
            type="text"
          />
        </div>
        <div
          class="sign-in__buttons"
        >
          <button
            class="button button--signin"
          >
            Sign in
          </button>
          <button
            class="button button--signin"
          >
            Sign in with Google
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
